File: services/backend/app/adapters/api/schemas/system.py
"""Pydantic schemas for system version and update endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class SystemVersionResponse(BaseModel):
    """Response for GET /api/v1/system/version."""

    current_version: str
    latest_version: str | None = None
    last_checked: float | None = None
    release_url: str | None = None


class UpdateResponse(BaseModel):
    """Response for POST /api/v1/system/update."""

    status: str  # "manual" | "started"
    mode: str = "manual"
    instructions: list[str] | None = None

File: services/frontend/src/api/system.ts
import { apiFetch } from './client'

export interface SystemVersionResponse {
  current_version: string
  latest_version: string | null
  last_checked: number | null
  release_url: string | null
}

export interface UpdateResponse {
  status: 'manual' | 'started' | 'ok' | 'error'
  mode: 'manual' | 'docker-socket'
  instructions: string[] | null
}

export function getVersion(refresh = false): Promise<SystemVersionResponse> {
  const qs = refresh ? '?refresh=true' : ''
  return apiFetch(`/api/v1/system/version${qs}`)
}

export function triggerUpdate(): Promise<UpdateResponse> {
  return apiFetch('/api/v1/system/update', { method: 'POST' })
}
